kontonummer.js

```diff
@@ -404,24 +409,38 @@
             return {
                 is_valid: false,
                 errors: errors,
-                matched_banks: matched_banks
+                matched_banks: matched_banks,
+                has_warnings: Boolean(warnings.length)
             };
         }
 
         return {
             is_valid: is_valid,
             errors: [],
-            matched_banks: matched_banks
+            matched_banks: matched_banks,
+            has_warnings: Boolean(warnings.length)
         };
     };
 
     var validateLength = function(bank, bankNumber) {
         var errors = [];
 
         if (bank.regex.test(bankNumber)) {
-            if (bankNumber.length < bank.lengths.clearing + bank.lengths.account) {
+            let actualBankNumber = bankNumber;
+            let minBankNumberLength =  bank.lengths.clearing + bank.lengths.account;
+            const maxBankNumberLength = bank.lengths.clearing + bank.lengths.account;
```

Comment:
This doesn't have a babel/ES6 compilation/build step, so we should aim to have it ES5 compatible, so no `const` or `let`.

```suggestion
            var maxBankNumberLength = bank.lengths.clearing + bank.lengths.account;
```